### PR TITLE
Hide RoomStatusBar when it displays nothing

### PR DESCRIPTION
**Pull request by lukebarnard1**
Notify RoomView of when RoomStatusBar has a size of '0' or not '0' via onHidden and onVisible props.

Should fix https://github.com/vector-im/riot-web/issues/1991 ?

### CODE REVIEW

**Comment by lukebarnard1** [comment]
This needs animation and debounce added to prevent jittering timeline

File: src/components/structures/RoomView.js [hunk 1]
```diff
@@ -1667,6 +1683,13 @@ module.exports = React.createClass({
                 </div>
             );
         }
+        let statusBarAreaClass = "mx_RoomView_statusArea mx_fadable mx_max_height_animated mx_margin_top_animated";
```
Comment (lukebarnard1) [inline src/components/structures/RoomView.js]: `mx_max_height_animated` and `mx_margin_top_animated` are not needed..

File: src/components/structures/RoomStatusBar.js [hunk 2]
```diff
@@ -20,6 +20,8 @@ var dis = require("../../dispatcher");
 var WhoIsTyping = require("../../WhoIsTyping");
 var MatrixClientPeg = require("../../MatrixClientPeg");
 
+const HIDE_DEBOUNCE_MS = 2000;
```
Comment (ara4n) [inline src/components/structures/RoomStatusBar.js]: i'd guess we want to wait at least 10s before the statusbar gets bored and hides itself again, but will have a play with it first...

File: src/components/structures/RoomStatusBar.js [hunk 3]
```diff
@@ -104,35 +125,24 @@ module.exports = React.createClass({
...
-        } else if (this.props.hasUnsentMessages) {
-            newSize = 2;
-        } else {
-            newSize = 0;
+    // We don't need the actual height - just whether it is likely to have
+    // changed - so we use '0' to indicate normal size, and other values to
+    // indicate other sizes.
+    _getSize: function(state, props) {
+        if (state.syncState === "ERROR" || state.whoisTypingString) {
+            return 1;
```
Comment (ara4n) [inline src/components/structures/RoomStatusBar.js]: whilst we're at it, can we get rid of these confusing numeric literals and use constants, just to make it clear we're using them as flags and they don't have numeric significance (e.g. ordering or dimensionality)?  Unless i'm misunderstanding what @richvdh was aiming for here?

**Comment by ara4n** [comment]
lgtm, other than the "not resetting when changing room" thing.

File: src/components/structures/RoomStatusBar.js (hunk 2 above)
Comment (lukebarnard1) [inline src/components/structures/RoomStatusBar.js]: I'm assuming that 2s felt ok then?

File: src/components/structures/RoomStatusBar.js (hunk 3 above)
Comment (lukebarnard1) [inline src/components/structures/RoomStatusBar.js]: Well they're kind-of dimensional, but in a topological way... So it's an ordering on size but not proportional to size.

Comment (richvdh) [inline src/components/structures/RoomStatusBar.js]: yup what @lukebarnard1 said. Could still use constants, though.

Comment (lukebarnard1) [inline src/components/structures/RoomStatusBar.js]: Constants it is!

File: src/components/structures/RoomStatusBar.js (hunk 2 above)
Comment (ara4n) [inline src/components/structures/RoomStatusBar.js]: no, it felt way too short to me - kept yoyoing like crazy.  Suggest 10s.

**Comment by ara4n** [comment]
lgtm other than timing tweak

**Pull request merged**